api/pkg/filtermanager/wrapper.go

```diff
@@ -129,11 +131,15 @@ func NewDebugFilter(name string, internal api.Filter, callbacks api.FilterCallba
 }
 
 func (f *debugFilter) recordExecution(start time.Time) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
 	duration := time.Since(start)
 	f.record += duration
 }
 
 func (f *debugFilter) reportExecution() (name string, duration time.Duration) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
```

Comment:
I don't have much concurrent programming experience, but can a read-write lock be used instead of a normal lock here?

Reply:
In current usage, the record is `write N read 1`. Only the `OnLog` phase will read the record:
https://github.com/mosn/htnn/blob/0d60076607d6551674828dcf99373b3c0b33176c/api/pkg/filtermanager/filtermanager.go#L988
because in other phases the record is incomplete.

Switching to RWMutex doesn't give performance improvement as it is more suitable for situations that read is much more than write. And unlike Mutex, the RWMutex doesn't have fast path which only uses atomic operation. In this use case, most time we will be on the fast path as few goroutines are still running in the `OnLog` phase.

Reply:
It make sense to me.
Thank your very much for your detailed explanation